pom.xml

```diff
@@ -177,10 +177,10 @@
     <connection>scm:git:git://github.com/Redouane59/twittered.git</connection>
     <developerConnection>scm:git:ssh://github.com:Redouane59/twittered.git
     </developerConnection>
-    <url>http://github.com/Redouane59/twittered</url>
+    <url>https://github.com/Redouane59/twittered</url>
   </scm>
 
-  <url>http://github.com/Redouane59/twittered</url>
+  <url>https://github.com/Redouane59/twittered</url>
 
-  <version>2.7</version>
+  <version>2.7-SNAPSHOT</version>
```

Comment:
Hey, I'm not really familiar with this. What does it change exactly to call it snapshot until the release ?
Currently I just releasing new versions when I have several PR to merge on master and they are currently always called 2.X. I'm not expert in production so i'm interested in your feedback :)

Reply:
A contributor will probably build the project multiple times using `mvn clean install` out of convience. The `install` will place the artifact in the local `.m2` repository. If we are using a full version identifier for a development version (not released), we can only have exactly one version at a time. We expect it to be equal. If we build a v2.7 today, it should be the same binary as if we build v2.7. tomorrow. 

The difference between a "real" version and a snapshot version is that snapshots might get updates. That means that downloading 1.0-SNAPSHOT today might give a different file than downloading it yesterday or tomorrow.

Usually, snapshot dependencies should only exist during development and no released version (i.e. no non-snapshot) should have a dependency on a snapshot version.

For a release, the SNAPSHOT is removed and we have the "real" version, which can then be pushed to Maven Central / Sonatype to be released.

Hope it helps?

Reply:
Perfect that's really clear. So once I start working on a new version, I should add the keyword `SNAPSHOT` until the day I release the final version. I'll do that from today, thanks !

Reply:
Glad, it helped.